records/2024-06-04.manifests-kustomize-cicd.md

```diff
@@ -0,0 +1,183 @@
+
+The manifests rollout workflow receives the new image tag and performs a manual patch of the existing deployment:
+
+```shell
+kubectl set image deployment.apps/api api=$DOCKER_SLUG:$DOCKER_TAG -n=notification-canada-ca
+```
+
+##### Admin Pull Request and Merge To Main
+
+![Staging Admin Current CICD](./diagrams/2024-06-04.manifests-current-cicd/current-admin-pr.png)
+
+##### API Pull Request and Merge To Main
+
+![Staging API Current CICD](./diagrams/2024-06-04.manifests-current-cicd/current-api-pr.png)
+
+##### Document Download API Pull Request and Merge To Main
+
+![Staging Document Download API Current CICD](./diagrams/2024-06-04.manifests-current-cicd/current-dd-api-pr.png)
+
+##### Documentation Pull Request and Merge To Main
+
+![Staging Documentation Current CICD](./diagrams/2024-06-04.manifests-current-cicd/current-documentation-pr.png)
+
+#### Kustomize/Helmfile/Manifests Pull Request and Merge To Main
+
+When a global manifests rollout is performed by the merge to main (staging) github workflow, all of the Kustomize code is deployed.
+
+```shell
+notification-manifests/env/staging kubectl apply -k .
+```
+
+![Staging Manifests Current CICD](./diagrams/2024-06-04.manifests-current-cicd/current-manifests-pr.png)
+
+##### Terraform Pull Request and Merge To Main
+
+![Staging Terraform Current CICD](./diagrams/2024-06-04.manifests-current-cicd/current-terraform-pr.png)
+
+### Production
+
+#### Individual Application Rollouts
+
+This concept does not exist for production. We only rollout all of Kustomize.
+
+#### Global Manifests Rollouts
+
+The Notify PR Bot generates an auto-pr on every merge to main in each of the application Github repositories (see above). The PR bot does not touch the helmfile side of things, so a global helmfile apply is done when a production release is executed.
+
+#### Kustomize/Helmfile/Manifests Production Release
+
+![Production Manifests Release](./diagrams/2024-06-04.manifests-current-cicd/current-manifests-pr.png)
+
+##### Terraform Production Release
+
+![Production Terraform Release](./diagrams/2024-06-04.manifests-current-cicd/current-terraform-release.png)
+
+### Problems
+
+#### Two Different Deployment Technologies
+
+We are currently managing two different Kubernetes deployment methods with Kustomize and Helmfile. It would be beneficial to have everything under one technology.
+
+#### Different Release Process For Staging and Production
+
+Manifests code is not actually changed when doing a release to staging, thus making it difficult to track changes and perform effective rollbacks.
+
+#### Different Release Process Between Terraform and Manifests
+
+Terraform does not use the notify-pr-bot to generate automatic PRs for Terraform changes. This creates unnecessary toil for the ops lead when doing Terraform releases.
```

Comment:
Plus helmfile, right? the pr-bot does not manage the helmfile and that makes it another release definition for the release.